vowpalwabbit/cats_pdf.cc

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+// Notes:
+// This reduction exists to be invoked as a top level reduction that
+// can ouput pdf related to cats_tree
+// It can also parse a continuous labeled example.
+
+#include "cats_pdf.h"
+#include "parse_args.h"
+#include "err_constants.h"
+#include "api_status.h"
+#include "cb_continuous_label.h"
+#include "debug_log.h"
+
+// Aliases
+using VW::LEARNER::single_learner;
+using std::endl;
+using VW::cb_continuous::continuous_label;
+using VW::cb_continuous::continuous_label_elm;
+using VW::config::make_option;
+using VW::config::option_group_definition;
+using VW::config::options_i;
+
+// Enable/Disable indented debug statements
+VW_DEBUG_ENABLE(false)
+
+// Forward declarations
```

Comment:
is this needed? what is cb_explore.cc for example including that it doesn't need to forward declare `finish_example` which is also used in that file?

Reply:
Forward declarations can be used to reduce dependency between modules.  This one can be eliminated by including a header file.  Only this symbol is needed so it's forward declared here.